yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -165,6 +165,7 @@ private[spark] class Client(
     val appContext = newApp.getApplicationSubmissionContext
     appContext.setApplicationName(args.appName)
     appContext.setQueue(args.amQueue)
+    appContext.setPriority(Priority.newInstance(args.priority))
```

Comment:
I believe they are going to have a default priority per queue setting so if we hardcode the default to 0 that might not mach what the queues default it.  I would rather leave this unset if not explicitly specified.